### ISSUE TITLE
Arguments for training DINO with ResNet50

### ISSUE THREAD

**Issue by mbsariyildiz** [facebookresearch/dino#133]
Hello! Congrats for the publication, and thank you for sharing your implementation.

I want to train DINO with ResNet50, but I am a bit confused about which arguments to set.
Because the arguments in this file
https://dl.fbaipublicfiles.com/dino/dino_resnet50_pretrain/args.txt
and the recommended arguments written in the README
https://github.com/facebookresearch/dino#resnet-50-and-other-convnets-trainings
don't match (like optimizer, weight decay), or am I looking at the wrong file/section of the readme?

Thanks!

---
Btw, the arguments file says "batch_size_per_gpu=51", I wonder how you ended up with 51 😄

**Comment by woctezuma** [comment]
Relevant: https://github.com/facebookresearch/dino/issues/100#issuecomment-898294378

**Comment by mathildecaron31** [comment]
Hi @mbsariyildiz thanks for your kind words.

The arguments in [this file](https://dl.fbaipublicfiles.com/dino/dino_resnet50_pretrain/args.txt) are exactly those that I used for the rn50 checkpoint released in this repo. The run is performed with large batch training (total batch is 4080 distributed over 80 gpus). If you do have access to enough gpus, I'd recommend you follow this setting.

Otherwise, if you run in a smaller batch setting, I think some arguments should be adapted. This is the setting that I tried to describe in the [readme](https://github.com/facebookresearch/dino#resnet-50-and-other-convnets-trainings) with a batch size of 512 (8 gpus *  64). I realized that I forgot the `--lr 0.03` in the command line btw. I will edit the readme.

----------
When launching jobs, I usually try to maximize the batch size that I can fit per gpu, hence the "batch_size_per_gpu=51". This way when distributing on 80 gpus I get a batch size of 4080. If I had been using a more standard "batch_size_per_gpu=32" for example I would have required 128 gpus to train with the same total batch size :).

Hope that helps.

**Comment by mathildecaron31** [comment]
7064315296736d16ae72c08194e9beb0d4b1c0fa

**Comment by mbsariyildiz** [comment]
Thank you very much @mathildecaron31 for the detailed answer! 
The difference between the two sets of arguments is clear now.
I'll try training ResNet50 with the arguments in the [readme](https://github.com/facebookresearch/dino#resnet-50-and-other-convnets-trainings) then.

**Comment by annawoodard** [comment]
> Btw, the arguments file says "batch_size_per_gpu=51", I wonder how you ended up with 51 😄

@mbsariyildiz in case it is of interest, there is a nice post [here](https://wandb.ai/datenzauberai/Batch-Size-Testing/reports/Do-Batch-Sizes-Actually-Need-to-be-Powers-of-2---VmlldzoyMDkwNDQx) showing that batch sizes in powers of 2 usually aren't necessary for minimizing training speed
